src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineControl.xaml.cs

```diff
@@ -0,0 +1,229 @@
+        {
+            InitializeComponent();
+
+            LanguageServiceBroker = languageServiceBroker;
+            ThreadingContext = threadingContext;
+            EditorAdaptersFactoryService = editorAdaptersFactoryService;
+            CodeWindow = codeWindow;
+            ComEventSink.Advise<IVsCodeWindowEvents>(codeWindow, this);
+            SortOption = SortOption.Order;
+
+            _computeUIModelQueue = new AsyncBatchingWorkQueue<bool, DocumentSymbolModel?>(
+                DelayTimeSpan.Short,
+                ComputeUIModelAsync,
+                EqualityComparer<bool>.Default,
+                asyncListener,
+                cancellationToken);
+
+            _updateUIModelQueue = new AsyncBatchingWorkQueue<bool, DocumentSymbolModel?>(
+                DelayTimeSpan.NearImmediate,
+                UpdateUIAsync,
+                EqualityComparer<bool>.Default,
+                asyncListener,
+                cancellationToken);
+
+            _highlightAndExpandNodesQueue = new AsyncBatchingWorkQueue<ExpansionOption>(
+                DelayTimeSpan.NearImmediate,
+                HightlightNodeAsync,
+                asyncListener,
+                cancellationToken);
+
+            _jumpToContentQueue = new AsyncBatchingWorkQueue<DocumentSymbolItem>(
+                DelayTimeSpan.NearImmediate,
+                JumpToContentAsync,
+                asyncListener,
+                cancellationToken);
+
+            // Primary text view is expected to exist on window initialization.
+            if (ErrorHandler.Failed(codeWindow.GetPrimaryView(out var primaryTextView)))
+                Debug.Fail("GetPrimaryView failed during DocumentOutlineControl initialization.");
+
+            if (ErrorHandler.Failed(StartTrackingView(primaryTextView)))
+                Debug.Fail("StartTrackingView failed during DocumentOutlineControl initialization.");
+
+            if (ErrorHandler.Succeeded(codeWindow.GetSecondaryView(out var secondaryTextView)))
+            {
+                if (ErrorHandler.Failed(StartTrackingView(secondaryTextView)))
+                    Debug.Fail("StartTrackingView failed during DocumentOutlineControl initialization.");
+            }
+
+            StartComputeUIModelTask();
+        }
+
+        int IVsCodeWindowEvents.OnNewView(IVsTextView pView)
+        {
+            ThreadingContext.ThrowIfNotOnUIThread();
+            return StartTrackingView(pView);
+        }
+
+        private int StartTrackingView(IVsTextView textView)
+        {
+            ThreadingContext.ThrowIfNotOnUIThread();
+
+            var wpfTextView = EditorAdaptersFactoryService.GetWpfTextView(textView);
+            if (wpfTextView is null)
+                return VSConstants.E_FAIL;
+
+            _trackedTextViews.Add(textView, wpfTextView);
+
+            wpfTextView.Caret.PositionChanged += Caret_PositionChanged;
+
+            // Subscribe only once since text buffer is the same for the primary and secondary text views.
```

Comment:
not sure i totally get what this comment means.  Also, why do we still subscrtibe to teh caret position twice in that case?  Can you doc why Caret registration is different from TextBuffer registration?  Thanks!

Reply:
Edited comment for clarity